.github/workflows/full-test-suite.yml

```diff
@@ -0,0 +1,23 @@
+name: full-test-suite
+on:
+  push:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.21.x"
+      - name: Install dependencies
+        run: go get .
+      - name: Build
+        run: make build
```

Comment:
No need to build the project twice. We can have a common build for both the unit tests and integration tests.

Reply:
Are you sure. I think there is a common Build. 
`make build` is only executed once before make unittest & make test. @JyotinderSingh

Reply:
Oh my bad, overlooked that.

Also do we still need the other unit test workflow file?

Reply:
Good call. I think workflow/file needs to be separate for triggers. Did not investigate too deeply but this was simplest to separate workflows.